src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -134,13 +153,16 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                     // Generate separate queries for each resource type and add them to query list.
                     foreach (string resourceType in _reindexJobRecord.Resources)
                     {
-                        var query = new ReindexJobQueryStatus(resourceType, continuationToken: null)
+                        if (_reindexJobRecord.ResourceCount[resourceType] > 0)
```

Comment:
>ResourceCount [](start = 46, length = 13)

Situation: We have lot of Patients, and no Organizations in DB. We start doing re-indexing and it takes day to go through all patients.
Somewhere in the middle of the day few organizations been posted.
Do we care about them?

Reply:
This is a really good question.

As discussed, any new resources created during a reindexing operation will be indexed with all search parameters marked as `supported` on creation, so it is by design that they aren't picked up here by the worker.